Doesn't compile on OS X

I don't know why, but latest commit doesn't compile successfully with latest XCode and Mac OS 10.13.4.
What I get:
```
> cryptonight-hashing@2.0.0 install /Users/andrey/Documents/Projects/xmr-node-proxy/node_modules/cryptonight-hashing
> node-gyp rebuild

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

  CXX(target) Release/obj.target/cryptonight-hashing/multihashing.o
In file included from ../multihashing.cc:13:
In file included from ../xmrig/crypto/CryptoNight_x86.h:30:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/x86intrin.h:29:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/immintrin.h:54:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/wmmintrin.h:29:
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/__wmmintrin_aes.h:45:27: error: 
      '__target__' attribute only applies to functions
static __inline__ __m128i __DEFAULT_FN_ATTRS
                          ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/__wmmintrin_aes.h:29:75: note: 
      expanded from macro '__DEFAULT_FN_ATTRS'
#define __DEFAULT_FN_ATTRS __attribute__((__always_inline__, __nodebug__, __targ...
                                                                          ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/__wmmintrin_aes.h:46:18: error: 
      redefinition of '__m128i' as different kind of symbol
_mm_aesenc_si128(__m128i __V, __m128i __R)
                 ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/emmintrin.h:30:19: note: 
      previous definition is here
typedef long long __m128i __attribute__((__vector_size__(16)));
                  ^
In file included from ../multihashing.cc:13:
In file included from ../xmrig/crypto/CryptoNight_x86.h:30:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/x86intrin.h:29:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/immintrin.h:54:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/wmmintrin.h:29:
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/__wmmintrin_aes.h:46:26: error: 
      expected ';' after top level declarator
_mm_aesenc_si128(__m128i __V, __m128i __R)
                         ^
In file included from ../multihashing.cc:13:
In file included from ../xmrig/crypto/CryptoNight_x86.h:30:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/x86intrin.h:29:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/immintrin.h:62:
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:71:28: error: 
      expected ')'
  return (__m256d)((__v4df)__a+(__v4df)__b);
                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:71:19: note: 
      to match this '('
  return (__m256d)((__v4df)__a+(__v4df)__b);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:71:21: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)((__v4df)__a+(__v4df)__b);
                    ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:107:28: error: 
      expected ')'
  return (__m256d)((__v4df)__a-(__v4df)__b);
                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:107:19: note: 
      to match this '('
  return (__m256d)((__v4df)__a-(__v4df)__b);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:107:21: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)((__v4df)__a-(__v4df)__b);
                    ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:144:47: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)__builtin_ia32_addsubpd256((__v4df)__a, (__v4df)__b);
                                              ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:181:28: error: 
      expected ')'
  return (__m256d)((__v4df)__a/(__v4df)__b);
                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:181:19: note: 
      to match this '('
  return (__m256d)((__v4df)__a/(__v4df)__b);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:181:21: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)((__v4df)__a/(__v4df)__b);
                    ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:218:44: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)__builtin_ia32_maxpd256((__v4df)__a, (__v4df)__b);
                                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:256:44: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)__builtin_ia32_minpd256((__v4df)__a, (__v4df)__b);
                                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:293:28: error: 
      expected ')'
  return (__m256d)((__v4df)__a * (__v4df)__b);
                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:293:19: note: 
      to match this '('
  return (__m256d)((__v4df)__a * (__v4df)__b);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:293:21: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)((__v4df)__a * (__v4df)__b);
                    ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:328:45: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)__builtin_ia32_sqrtpd256((__v4df)__a);
                                            ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:687:45: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)__builtin_ia32_haddpd256((__v4df)__a, (__v4df)__b);
                                            ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:733:45: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)__builtin_ia32_hsubpd256((__v4df)__a, (__v4df)__b);
                                            ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:825:51: error: 
      use of undeclared identifier '__v4df'
  return (__m256d)__builtin_ia32_vpermilvarpd256((__v4df)__a, (__v4di)__c);
                                                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:1426:6: error: 
      use of undeclared identifier '__v4df'
    (__v4df)__a, (__v4df)__b, (__v4df)__c);
     ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.

```

I tried commit https://github.com/MoneroOcean/node-cryptonight-hashing/commit/a3ac72f82cf0e370de98c3278020fd3eb67cb6bd and it works fine. Any ideas?

Yes, commit https://github.com/MoneroOcean/node-cryptonight-hashing/commit/f85a0c9f02dd6e09c919a9d9f24c2757c142744b make troubles. 
```
Andreys-iMac:temp andrey$ git clone https://github.com/MoneroOcean/node-cryptonight-hashing
Cloning into 'node-cryptonight-hashing'...
remote: Counting objects: 660, done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 660 (delta 5), reused 12 (delta 3), pack-reused 644
Receiving objects: 100% (660/660), 1.20 MiB | 1.12 MiB/s, done.
Resolving deltas: 100% (316/316), done.
Andreys-iMac:temp andrey$ cd node*
Andreys-iMac:node-cryptonight-hashing andrey$ git checkout 918580a
Note: checking out '918580a'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 918580a Fixed example
Andreys-iMac:node-cryptonight-hashing andrey$ clang -dM -E -x c /dev/null
#define OBJC_NEW_PROPERTIES 1
#define _LP64 1
#define __APPLE_CC__ 6000
#define __APPLE__ 1
#define __ATOMIC_ACQUIRE 2
#define __ATOMIC_ACQ_REL 4
#define __ATOMIC_CONSUME 1
#define __ATOMIC_RELAXED 0
#define __ATOMIC_RELEASE 3
#define __ATOMIC_SEQ_CST 5
#define __BIGGEST_ALIGNMENT__ 16
#define __BLOCKS__ 1
#define __BYTE_ORDER__ __ORDER_LITTLE_ENDIAN__
#define __CHAR16_TYPE__ unsigned short
#define __CHAR32_TYPE__ unsigned int
#define __CHAR_BIT__ 8
#define __CLANG_ATOMIC_BOOL_LOCK_FREE 2
#define __CLANG_ATOMIC_CHAR16_T_LOCK_FREE 2
#define __CLANG_ATOMIC_CHAR32_T_LOCK_FREE 2
#define __CLANG_ATOMIC_CHAR_LOCK_FREE 2
#define __CLANG_ATOMIC_INT_LOCK_FREE 2
#define __CLANG_ATOMIC_LLONG_LOCK_FREE 2
#define __CLANG_ATOMIC_LONG_LOCK_FREE 2
#define __CLANG_ATOMIC_POINTER_LOCK_FREE 2
#define __CLANG_ATOMIC_SHORT_LOCK_FREE 2
#define __CLANG_ATOMIC_WCHAR_T_LOCK_FREE 2
#define __CONSTANT_CFSTRINGS__ 1
#define __DBL_DECIMAL_DIG__ 17
#define __DBL_DENORM_MIN__ 4.9406564584124654e-324
#define __DBL_DIG__ 15
#define __DBL_EPSILON__ 2.2204460492503131e-16
#define __DBL_HAS_DENORM__ 1
#define __DBL_HAS_INFINITY__ 1
#define __DBL_HAS_QUIET_NAN__ 1
#define __DBL_MANT_DIG__ 53
#define __DBL_MAX_10_EXP__ 308
#define __DBL_MAX_EXP__ 1024
#define __DBL_MAX__ 1.7976931348623157e+308
#define __DBL_MIN_10_EXP__ (-307)
#define __DBL_MIN_EXP__ (-1021)
#define __DBL_MIN__ 2.2250738585072014e-308
#define __DECIMAL_DIG__ __LDBL_DECIMAL_DIG__
#define __DYNAMIC__ 1
#define __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ 101300
#define __FINITE_MATH_ONLY__ 0
#define __FLT_DECIMAL_DIG__ 9
#define __FLT_DENORM_MIN__ 1.40129846e-45F
#define __FLT_DIG__ 6
#define __FLT_EPSILON__ 1.19209290e-7F
#define __FLT_EVAL_METHOD__ 0
#define __FLT_HAS_DENORM__ 1
#define __FLT_HAS_INFINITY__ 1
#define __FLT_HAS_QUIET_NAN__ 1
#define __FLT_MANT_DIG__ 24
#define __FLT_MAX_10_EXP__ 38
#define __FLT_MAX_EXP__ 128
#define __FLT_MAX__ 3.40282347e+38F
#define __FLT_MIN_10_EXP__ (-37)
#define __FLT_MIN_EXP__ (-125)
#define __FLT_MIN__ 1.17549435e-38F
#define __FLT_RADIX__ 2
#define __FXSR__ 1
#define __GCC_ATOMIC_BOOL_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR16_T_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR32_T_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR_LOCK_FREE 2
#define __GCC_ATOMIC_INT_LOCK_FREE 2
#define __GCC_ATOMIC_LLONG_LOCK_FREE 2
#define __GCC_ATOMIC_LONG_LOCK_FREE 2
#define __GCC_ATOMIC_POINTER_LOCK_FREE 2
#define __GCC_ATOMIC_SHORT_LOCK_FREE 2
#define __GCC_ATOMIC_TEST_AND_SET_TRUEVAL 1
#define __GCC_ATOMIC_WCHAR_T_LOCK_FREE 2
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_16 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8 1
#define __GNUC_MINOR__ 2
#define __GNUC_PATCHLEVEL__ 1
#define __GNUC_STDC_INLINE__ 1
#define __GNUC__ 4
#define __GXX_ABI_VERSION 1002
#define __INT16_C_SUFFIX__ 
#define __INT16_FMTd__ "hd"
#define __INT16_FMTi__ "hi"
#define __INT16_MAX__ 32767
#define __INT16_TYPE__ short
#define __INT32_C_SUFFIX__ 
#define __INT32_FMTd__ "d"
#define __INT32_FMTi__ "i"
#define __INT32_MAX__ 2147483647
#define __INT32_TYPE__ int
#define __INT64_C_SUFFIX__ LL
#define __INT64_FMTd__ "lld"
#define __INT64_FMTi__ "lli"
#define __INT64_MAX__ 9223372036854775807LL
#define __INT64_TYPE__ long long int
#define __INT8_C_SUFFIX__ 
#define __INT8_FMTd__ "hhd"
#define __INT8_FMTi__ "hhi"
#define __INT8_MAX__ 127
#define __INT8_TYPE__ signed char
#define __INTMAX_C_SUFFIX__ L
#define __INTMAX_FMTd__ "ld"
#define __INTMAX_FMTi__ "li"
#define __INTMAX_MAX__ 9223372036854775807L
#define __INTMAX_TYPE__ long int
#define __INTMAX_WIDTH__ 64
#define __INTPTR_FMTd__ "ld"
#define __INTPTR_FMTi__ "li"
#define __INTPTR_MAX__ 9223372036854775807L
#define __INTPTR_TYPE__ long int
#define __INTPTR_WIDTH__ 64
#define __INT_FAST16_FMTd__ "hd"
#define __INT_FAST16_FMTi__ "hi"
#define __INT_FAST16_MAX__ 32767
#define __INT_FAST16_TYPE__ short
#define __INT_FAST32_FMTd__ "d"
#define __INT_FAST32_FMTi__ "i"
#define __INT_FAST32_MAX__ 2147483647
#define __INT_FAST32_TYPE__ int
#define __INT_FAST64_FMTd__ "ld"
#define __INT_FAST64_FMTi__ "li"
#define __INT_FAST64_MAX__ 9223372036854775807L
#define __INT_FAST64_TYPE__ long int
#define __INT_FAST8_FMTd__ "hhd"
#define __INT_FAST8_FMTi__ "hhi"
#define __INT_FAST8_MAX__ 127
#define __INT_FAST8_TYPE__ signed char
#define __INT_LEAST16_FMTd__ "hd"
#define __INT_LEAST16_FMTi__ "hi"
#define __INT_LEAST16_MAX__ 32767
#define __INT_LEAST16_TYPE__ short
#define __INT_LEAST32_FMTd__ "d"
#define __INT_LEAST32_FMTi__ "i"
#define __INT_LEAST32_MAX__ 2147483647
#define __INT_LEAST32_TYPE__ int
#define __INT_LEAST64_FMTd__ "ld"
#define __INT_LEAST64_FMTi__ "li"
#define __INT_LEAST64_MAX__ 9223372036854775807L
#define __INT_LEAST64_TYPE__ long int
#define __INT_LEAST8_FMTd__ "hhd"
#define __INT_LEAST8_FMTi__ "hhi"
#define __INT_LEAST8_MAX__ 127
#define __INT_LEAST8_TYPE__ signed char
#define __INT_MAX__ 2147483647
#define __LDBL_DECIMAL_DIG__ 21
#define __LDBL_DENORM_MIN__ 3.64519953188247460253e-4951L
#define __LDBL_DIG__ 18
#define __LDBL_EPSILON__ 1.08420217248550443401e-19L
#define __LDBL_HAS_DENORM__ 1
#define __LDBL_HAS_INFINITY__ 1
#define __LDBL_HAS_QUIET_NAN__ 1
#define __LDBL_MANT_DIG__ 64
#define __LDBL_MAX_10_EXP__ 4932
#define __LDBL_MAX_EXP__ 16384
#define __LDBL_MAX__ 1.18973149535723176502e+4932L
#define __LDBL_MIN_10_EXP__ (-4931)
#define __LDBL_MIN_EXP__ (-16381)
#define __LDBL_MIN__ 3.36210314311209350626e-4932L
#define __LITTLE_ENDIAN__ 1
#define __LONG_LONG_MAX__ 9223372036854775807LL
#define __LONG_MAX__ 9223372036854775807L
#define __LP64__ 1
#define __MACH__ 1
#define __MMX__ 1
#define __NO_INLINE__ 1
#define __NO_MATH_INLINES 1
#define __OBJC_BOOL_IS_BOOL 0
#define __ORDER_BIG_ENDIAN__ 4321
#define __ORDER_LITTLE_ENDIAN__ 1234
#define __ORDER_PDP_ENDIAN__ 3412
#define __PIC__ 2
#define __POINTER_WIDTH__ 64
#define __PRAGMA_REDEFINE_EXTNAME 1
#define __PTRDIFF_FMTd__ "ld"
#define __PTRDIFF_FMTi__ "li"
#define __PTRDIFF_MAX__ 9223372036854775807L
#define __PTRDIFF_TYPE__ long int
#define __PTRDIFF_WIDTH__ 64
#define __REGISTER_PREFIX__ 
#define __SCHAR_MAX__ 127
#define __SHRT_MAX__ 32767
#define __SIG_ATOMIC_MAX__ 2147483647
#define __SIG_ATOMIC_WIDTH__ 32
#define __SIZEOF_DOUBLE__ 8
#define __SIZEOF_FLOAT__ 4
#define __SIZEOF_INT128__ 16
#define __SIZEOF_INT__ 4
#define __SIZEOF_LONG_DOUBLE__ 16
#define __SIZEOF_LONG_LONG__ 8
#define __SIZEOF_LONG__ 8
#define __SIZEOF_POINTER__ 8
#define __SIZEOF_PTRDIFF_T__ 8
#define __SIZEOF_SHORT__ 2
#define __SIZEOF_SIZE_T__ 8
#define __SIZEOF_WCHAR_T__ 4
#define __SIZEOF_WINT_T__ 4
#define __SIZE_FMTX__ "lX"
#define __SIZE_FMTo__ "lo"
#define __SIZE_FMTu__ "lu"
#define __SIZE_FMTx__ "lx"
#define __SIZE_MAX__ 18446744073709551615UL
#define __SIZE_TYPE__ long unsigned int
#define __SIZE_WIDTH__ 64
#define __SSE2_MATH__ 1
#define __SSE2__ 1
#define __SSE3__ 1
#define __SSE4_1__ 1
#define __SSE_MATH__ 1
#define __SSE__ 1
#define __SSP__ 1
#define __SSSE3__ 1
#define __STDC_HOSTED__ 1
#define __STDC_NO_THREADS__ 1
#define __STDC_UTF_16__ 1
#define __STDC_UTF_32__ 1
#define __STDC_VERSION__ 201112L
#define __STDC__ 1
#define __UINT16_C_SUFFIX__ 
#define __UINT16_FMTX__ "hX"
#define __UINT16_FMTo__ "ho"
#define __UINT16_FMTu__ "hu"
#define __UINT16_FMTx__ "hx"
#define __UINT16_MAX__ 65535
#define __UINT16_TYPE__ unsigned short
#define __UINT32_C_SUFFIX__ U
#define __UINT32_FMTX__ "X"
#define __UINT32_FMTo__ "o"
#define __UINT32_FMTu__ "u"
#define __UINT32_FMTx__ "x"
#define __UINT32_MAX__ 4294967295U
#define __UINT32_TYPE__ unsigned int
#define __UINT64_C_SUFFIX__ ULL
#define __UINT64_FMTX__ "llX"
#define __UINT64_FMTo__ "llo"
#define __UINT64_FMTu__ "llu"
#define __UINT64_FMTx__ "llx"
#define __UINT64_MAX__ 18446744073709551615ULL
#define __UINT64_TYPE__ long long unsigned int
#define __UINT8_C_SUFFIX__ 
#define __UINT8_FMTX__ "hhX"
#define __UINT8_FMTo__ "hho"
#define __UINT8_FMTu__ "hhu"
#define __UINT8_FMTx__ "hhx"
#define __UINT8_MAX__ 255
#define __UINT8_TYPE__ unsigned char
#define __UINTMAX_C_SUFFIX__ UL
#define __UINTMAX_FMTX__ "lX"
#define __UINTMAX_FMTo__ "lo"
#define __UINTMAX_FMTu__ "lu"
#define __UINTMAX_FMTx__ "lx"
#define __UINTMAX_MAX__ 18446744073709551615UL
#define __UINTMAX_TYPE__ long unsigned int
#define __UINTMAX_WIDTH__ 64
#define __UINTPTR_FMTX__ "lX"
#define __UINTPTR_FMTo__ "lo"
#define __UINTPTR_FMTu__ "lu"
#define __UINTPTR_FMTx__ "lx"
#define __UINTPTR_MAX__ 18446744073709551615UL
#define __UINTPTR_TYPE__ long unsigned int
#define __UINTPTR_WIDTH__ 64
#define __UINT_FAST16_FMTX__ "hX"
#define __UINT_FAST16_FMTo__ "ho"
#define __UINT_FAST16_FMTu__ "hu"
#define __UINT_FAST16_FMTx__ "hx"
#define __UINT_FAST16_MAX__ 65535
#define __UINT_FAST16_TYPE__ unsigned short
#define __UINT_FAST32_FMTX__ "X"
#define __UINT_FAST32_FMTo__ "o"
#define __UINT_FAST32_FMTu__ "u"
#define __UINT_FAST32_FMTx__ "x"
#define __UINT_FAST32_MAX__ 4294967295U
#define __UINT_FAST32_TYPE__ unsigned int
#define __UINT_FAST64_FMTX__ "lX"
#define __UINT_FAST64_FMTo__ "lo"
#define __UINT_FAST64_FMTu__ "lu"
#define __UINT_FAST64_FMTx__ "lx"
#define __UINT_FAST64_MAX__ 18446744073709551615UL
#define __UINT_FAST64_TYPE__ long unsigned int
#define __UINT_FAST8_FMTX__ "hhX"
#define __UINT_FAST8_FMTo__ "hho"
#define __UINT_FAST8_FMTu__ "hhu"
#define __UINT_FAST8_FMTx__ "hhx"
#define __UINT_FAST8_MAX__ 255
#define __UINT_FAST8_TYPE__ unsigned char
#define __UINT_LEAST16_FMTX__ "hX"
#define __UINT_LEAST16_FMTo__ "ho"
#define __UINT_LEAST16_FMTu__ "hu"
#define __UINT_LEAST16_FMTx__ "hx"
#define __UINT_LEAST16_MAX__ 65535
#define __UINT_LEAST16_TYPE__ unsigned short
#define __UINT_LEAST32_FMTX__ "X"
#define __UINT_LEAST32_FMTo__ "o"
#define __UINT_LEAST32_FMTu__ "u"
#define __UINT_LEAST32_FMTx__ "x"
#define __UINT_LEAST32_MAX__ 4294967295U
#define __UINT_LEAST32_TYPE__ unsigned int
#define __UINT_LEAST64_FMTX__ "lX"
#define __UINT_LEAST64_FMTo__ "lo"
#define __UINT_LEAST64_FMTu__ "lu"
#define __UINT_LEAST64_FMTx__ "lx"
#define __UINT_LEAST64_MAX__ 18446744073709551615UL
#define __UINT_LEAST64_TYPE__ long unsigned int
#define __UINT_LEAST8_FMTX__ "hhX"
#define __UINT_LEAST8_FMTo__ "hho"
#define __UINT_LEAST8_FMTu__ "hhu"
#define __UINT_LEAST8_FMTx__ "hhx"
#define __UINT_LEAST8_MAX__ 255
#define __UINT_LEAST8_TYPE__ unsigned char
#define __USER_LABEL_PREFIX__ _
#define __VERSION__ "4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)"
#define __WCHAR_MAX__ 2147483647
#define __WCHAR_TYPE__ int
#define __WCHAR_WIDTH__ 32
#define __WINT_TYPE__ int
#define __WINT_WIDTH__ 32
#define __amd64 1
#define __amd64__ 1
#define __apple_build_version__ 9020039
#define __block __attribute__((__blocks__(byref)))
#define __clang__ 1
#define __clang_major__ 9
#define __clang_minor__ 1
#define __clang_patchlevel__ 0
#define __clang_version__ "9.1.0 (clang-902.0.39.2)"
#define __core2 1
#define __core2__ 1
#define __llvm__ 1
#define __nonnull _Nonnull
#define __null_unspecified _Null_unspecified
#define __nullable _Nullable
#define __pic__ 2
#define __strong 
#define __tune_core2__ 1
#define __unsafe_unretained 
#define __weak __attribute__((objc_gc(weak)))
#define __x86_64 1
#define __x86_64__ 1
Andreys-iMac:node-cryptonight-hashing andrey$ npm install

> cryptonight-hashing@2.0.0 install /Users/andrey/Documents/temp/node-cryptonight-hashing
> node-gyp rebuild

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

  CXX(target) Release/obj.target/cryptonight-hashing/multihashing.o
../multihashing.cc:12:2: warning: Using software AES [-W#warnings]
#warning Using software AES
 ^
../multihashing.cc:159:23: warning: 'Call' is deprecated
      [-Wdeprecated-declarations]
            callback->Call(2, argv);
                      ^
../node_modules/nan/nan.h:1617:3: note: 'Call' has been explicitly marked
      deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../node_modules/nan/nan.h:98:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../multihashing.cc:226:23: warning: 'Call' is deprecated
      [-Wdeprecated-declarations]
            callback->Call(2, argv);
                      ^
../node_modules/nan/nan.h:1617:3: note: 'Call' has been explicitly marked
      deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../node_modules/nan/nan.h:98:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../multihashing.cc:289:23: warning: 'Call' is deprecated
      [-Wdeprecated-declarations]
            callback->Call(2, argv);
                      ^
../node_modules/nan/nan.h:1617:3: note: 'Call' has been explicitly marked
      deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../node_modules/nan/nan.h:98:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
4 warnings generated.
  CC(target) Release/obj.target/cryptonight-hashing/xmrig/crypto/c_blake256.o
  CC(target) Release/obj.target/cryptonight-hashing/xmrig/crypto/c_groestl.o
../xmrig/crypto/c_groestl.c:210:9: warning: comparison of integers of different
      signs: 'int' and 'unsigned long' [-Wsign-compare]
  for(;i<(SIZE512/sizeof(uint32_t));i++)
       ~^ ~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
  CC(target) Release/obj.target/cryptonight-hashing/xmrig/crypto/c_jh.o
  CC(target) Release/obj.target/cryptonight-hashing/xmrig/crypto/c_skein.o
  CXX(target) Release/obj.target/cryptonight-hashing/xmrig/common/crypto/keccak.o
  SOLINK_MODULE(target) Release/cryptonight-hashing.node
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN cryptonight-hashing@2.0.0 No license field.

added 2 packages in 6.39s
Andreys-iMac:node-cryptonight-hashing andrey$ git checkout master
M	package.json
Previous HEAD position was 918580a Fixed example
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
Andreys-iMac:node-cryptonight-hashing andrey$ npm install

> cryptonight-hashing@2.0.0 install /Users/andrey/Documents/temp/node-cryptonight-hashing
> node-gyp rebuild

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

  CXX(target) Release/obj.target/cryptonight-hashing/multihashing.o
In file included from ../multihashing.cc:13:
In file included from ../xmrig/crypto/CryptoNight_x86.h:30:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/x86intrin.h:29:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/immintrin.h:54:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/wmmintrin.h:29:
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/__wmmintrin_aes.h:45:27: error: '__target__'
      attribute only applies to functions
static __inline__ __m128i __DEFAULT_FN_ATTRS
                          ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/__wmmintrin_aes.h:29:75: note: expanded from
      macro '__DEFAULT_FN_ATTRS'
#define __DEFAULT_FN_ATTRS __attribute__((__always_inline__, __nodebug__, __target__("aes")))
                                                                          ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/__wmmintrin_aes.h:46:18: error: redefinition of
      '__m128i' as different kind of symbol
_mm_aesenc_si128(__m128i __V, __m128i __R)
                 ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/emmintrin.h:30:19: note: previous definition is
      here
typedef long long __m128i __attribute__((__vector_size__(16)));
                  ^
In file included from ../multihashing.cc:13:
In file included from ../xmrig/crypto/CryptoNight_x86.h:30:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/x86intrin.h:29:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/immintrin.h:54:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/wmmintrin.h:29:
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/__wmmintrin_aes.h:46:26: error: expected ';'
      after top level declarator
_mm_aesenc_si128(__m128i __V, __m128i __R)
                         ^
In file included from ../multihashing.cc:13:
In file included from ../xmrig/crypto/CryptoNight_x86.h:30:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/x86intrin.h:29:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/immintrin.h:62:
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:71:28: error: expected ')'
  return (__m256d)((__v4df)__a+(__v4df)__b);
                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:71:19: note: to match this '('
  return (__m256d)((__v4df)__a+(__v4df)__b);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:71:21: error: use of undeclared
      identifier '__v4df'
  return (__m256d)((__v4df)__a+(__v4df)__b);
                    ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:107:28: error: expected ')'
  return (__m256d)((__v4df)__a-(__v4df)__b);
                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:107:19: note: to match this '('
  return (__m256d)((__v4df)__a-(__v4df)__b);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:107:21: error: use of undeclared
      identifier '__v4df'
  return (__m256d)((__v4df)__a-(__v4df)__b);
                    ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:144:47: error: use of undeclared
      identifier '__v4df'
  return (__m256d)__builtin_ia32_addsubpd256((__v4df)__a, (__v4df)__b);
                                              ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:181:28: error: expected ')'
  return (__m256d)((__v4df)__a/(__v4df)__b);
                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:181:19: note: to match this '('
  return (__m256d)((__v4df)__a/(__v4df)__b);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:181:21: error: use of undeclared
      identifier '__v4df'
  return (__m256d)((__v4df)__a/(__v4df)__b);
                    ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:218:44: error: use of undeclared
      identifier '__v4df'
  return (__m256d)__builtin_ia32_maxpd256((__v4df)__a, (__v4df)__b);
                                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:256:44: error: use of undeclared
      identifier '__v4df'
  return (__m256d)__builtin_ia32_minpd256((__v4df)__a, (__v4df)__b);
                                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:293:28: error: expected ')'
  return (__m256d)((__v4df)__a * (__v4df)__b);
                           ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:293:19: note: to match this '('
  return (__m256d)((__v4df)__a * (__v4df)__b);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:293:21: error: use of undeclared
      identifier '__v4df'
  return (__m256d)((__v4df)__a * (__v4df)__b);
                    ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:328:45: error: use of undeclared
      identifier '__v4df'
  return (__m256d)__builtin_ia32_sqrtpd256((__v4df)__a);
                                            ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:687:45: error: use of undeclared
      identifier '__v4df'
  return (__m256d)__builtin_ia32_haddpd256((__v4df)__a, (__v4df)__b);
                                            ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:733:45: error: use of undeclared
      identifier '__v4df'
  return (__m256d)__builtin_ia32_hsubpd256((__v4df)__a, (__v4df)__b);
                                            ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:825:51: error: use of undeclared
      identifier '__v4df'
  return (__m256d)__builtin_ia32_vpermilvarpd256((__v4df)__a, (__v4di)__c);
                                                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.1.0/include/avxintrin.h:1426:6: error: use of undeclared
      identifier '__v4df'
    (__v4df)__a, (__v4df)__b, (__v4df)__c);
     ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/cryptonight-hashing/multihashing.o] Error 1
```
I've created a PR which resolves that: https://github.com/MoneroOcean/node-cryptonight-hashing/pull/4

--- a/multihashing.cc
+++ b/multihashing.cc
@@ -5,7 +5,7 @@
 #include <nan.h>
 
 #if defined(__AES__) && (__AES__ == 1)
-#else
+#elseif !defined(__APPLE__)
 #define _mm_aeskeygenassist_si128(a, b) a
 #define _mm_aesenc_si128(a, b) a
 #endif

